src/lib/prepareOas.ts

```diff
@@ -1,45 +1,107 @@
-import fs from 'fs';
-
+import ciDetect from '@npmcli/ci-detect';
 import chalk from 'chalk';
 import OASNormalize from 'oas-normalize';
 import ora from 'ora';
+import prompts from 'prompts';
 
 import { debug, info, oraOptions } from './logger';
+import readdirRecursive from './readdirRecursive';
+
+type FileSelection = {
+  file: string;
+};
 
 /**
  * Normalizes, validates, and (optionally) bundles an OpenAPI definition.
  *
  * @param {String} path path to spec file. if this is missing, the current directory is searched
  * for certain file names
- * @param {('openapi'|'validate')} command string to distinguish if it's being run in
+ * @param command string to distinguish if it's being run in
  * an 'openapi' or 'validate' context
  */
 export default async function prepareOas(path: string, command: 'openapi' | 'validate') {
   let specPath = path;
 
   if (!specPath) {
-    // If the user didn't supply an API specification, let's try to locate what they've got, and validate that. If they
-    // don't have any, let's let the user know how they can get one going.
-    specPath = await new Promise((resolve, reject) => {
-      ['swagger.json', 'swagger.yaml', 'swagger.yml', 'openapi.json', 'openapi.yaml', 'openapi.yml'].forEach(file => {
-        debug(`looking for definition with filename: ${file}`);
-        if (!fs.existsSync(file)) {
-          debug(`${file} not found`);
-          return;
-        }
-
-        info(
-          chalk.yellow(`We found ${file} and are attempting to ${command === 'openapi' ? 'upload' : 'validate'} it.`)
-        );
-        resolve(file);
+    /**
+     * Scans working directory for a potential OpenAPI or Swagger file.
+     * Any files in the `.git` directory or defined in a top-level `.gitignore` file
+     * are skipped.
+     *
+     * A "potential OpenAPI or Swagger file" is defined as a YAML or JSON file
+     * that has an `openapi` or `swagger` property defined at the top-level.
+     *
+     * If multiple potential files are found, the user must select a single file.
+     *
+     * An error is thrown in the following cases:
+     * - if in a CI environment and multiple files are found
+     * - no files are found
+     */
+
+    const fileFindingSpinner = ora({ text: 'Attempting to locate API definitions...', ...oraOptions() }).start();
+
+    const action = command === 'openapi' ? 'upload' : 'validate';
+
+    const jsonAndYamlFiles = readdirRecursive('.', true).filter(
+      file =>
+        file.toLowerCase().endsWith('.json') ||
+        file.toLowerCase().endsWith('.yaml') ||
+        file.toLowerCase().endsWith('.yml')
+    );
+
+    debug(`number of JSON or YAML files found: ${jsonAndYamlFiles.length}`);
+
+    const possibleSpecFiles = (
+      await Promise.all(
+        jsonAndYamlFiles.map(async file => {
```

Comment:
This need to be async?

```suggestion
        jsonAndYamlFiles.map(file => {
```

Reply:
ah damn left over from a previous refactor, nice catch

Reply:
wish TS or ESLint would catch this

Reply:
yeah same, the rule we have in place only catches them on non-closure declarations. kind of annoying.